src/network/mcpe/handler/ItemStackRequestExecutor.php

```diff
@@ -333,11 +335,20 @@ protected function processItemStackRequestAction(ItemStackRequestAction $action)
 
 			$this->setNextCreatedItem($item, true);
 		}elseif($action instanceof CraftRecipeStackRequestAction){
-			$this->beginCrafting($action->getRecipeId(), 1);
+			$recipeId = $action->getRecipeId();
+			$window = $this->player->getCurrentWindow();
+			if($window instanceof EnchantInventory){
+				$this->specialTransaction = new EnchantTransaction($this->player, $recipeId);
+				$this->setNextCreatedItem($window->getOutput($recipeId));
+				return;
+			}
+			$this->beginCrafting($recipeId, 1);
 		}elseif($action instanceof CraftRecipeAutoStackRequestAction){
 			$this->beginCrafting($action->getRecipeId(), $action->getRepetitions());
 		}elseif($action instanceof CraftingConsumeInputStackRequestAction){
-			$this->assertDoingCrafting();
+			if(!$this->specialTransaction instanceof EnchantTransaction){
+				$this->assertDoingCrafting();
+			}
```

Comment:
`assertDoingCrafting()` should probably be changed here instead

Reply:
That's what I did in the beginning, but `assertDoingCrafting` is called by the CraftingCreateSpecificResultStackRequestAction, which doesn't occur when enchanting. Is that okay?